### DISCUSSION TITLE
How to configure WireGuard Easy in full mesh mode

### DISCUSSION THREAD

**Discussion by xd19861224** [wg-easy/wg-easy#1046] [Ideas]
How to configure WireGuard Easy in full mesh mode

**Comment by eandersons** [comment]
Is this a question (How to configure WireGuard Easy in full mesh mode?) or unfinished instruction (titled "How to configure WireGuard Easy in full mesh mode")? The intention can be misinterpreted without proper punctuation and context.

Nitpicking aside, WireGuard Easy does not provide such functionality. For that you would have to use more advanced WireGuard management tool (for options see [Mesh Network by cedrickchee/awesome-wireguard](https://github.com/cedrickchee/awesome-wireguard?tab=readme-ov-file#mesh-network) and [HarvsG/WireGuardMeshes](https://github.com/HarvsG/WireGuardMeshes)).

**Comment by okfro** [comment]
I think it does get very complicated, very quickly to try to automatically create a mesh config that will just work. In my experience joining two VPS sites, a homelab and a road warrior, I have to manually edit a lot of the configs.

What wg-easy could do that would be extremely helpful to me would be to have the option to generate the client configs with all peers (basically, include every other client). I need to manually edit the port and the endpoint for each client--and the endpoint can be different depending on the client; but it would be sufficient to just be able to specify the public ip endpoint:port in wg-easy and then manually edit the configs where necessary.

Anyways, to boil that down into a concrete ask: would it be possible to:

- Add optional fields to Client edit form:
  - Client Port
  - Endpoint IP address
  - Endpoint Port
  - Raw Config (text field to store mesh config)
- Add "Build Mesh" button to the Clients grid (Advanced/Experimental option)
  - Clicking "Build Mesh" will generate the config for each client and store it on the Raw Config slot. 
  - Clicking "Build Mesh" will show a warning that any edits to client's raw config will be lost when the mesh config is regenerated.
- Downloading the client config will pull the raw config if available.

I think this would get many people 80% there in terms of having a functional mesh out of the box. Since everyone has different setups, there is no one-size fits all solution, but this would save a huge amount of time and energy and go a long way to making wg-easy my 3/4 stop solution.
